service/history/replication/eager_namespace_refresher.go

```diff
@@ -29,37 +29,63 @@ import (
 	"sync"
 
 	"go.temporal.io/api/serviceerror"
+	"go.temporal.io/server/api/adminservice/v1"
+	enumsspb "go.temporal.io/server/api/enums/v1"
+	replicationspb "go.temporal.io/server/api/replication/v1"
+	"go.temporal.io/server/client"
 	"go.temporal.io/server/common/headers"
 	"go.temporal.io/server/common/log"
 	"go.temporal.io/server/common/log/tag"
+	"go.temporal.io/server/common/metrics"
 	"go.temporal.io/server/common/namespace"
 	"go.temporal.io/server/common/persistence"
 )
 
 type (
 	EagerNamespaceRefresher interface {
 		UpdateNamespaceFailoverVersion(namespaceId namespace.ID, targetFailoverVersion int64) error
+		SyncNamespaceFromSourceCluster(ctx context.Context, namespaceId namespace.ID, sourceCluster string) error
 	}
 
 	eagerNamespaceRefresherImpl struct {
-		metadataManager   persistence.MetadataManager
-		namespaceRegistry namespace.Registry
-		logger            log.Logger
-		lock              sync.Mutex
+		metadataManager         persistence.MetadataManager
+		namespaceRegistry       namespace.Registry
+		logger                  log.Logger
+		lock                    sync.Mutex
+		clientBean              client.Bean
+		replicationTaskExecutor namespace.ReplicationTaskExecutor
+		currentCluster          string
+		metricsHandler          metrics.Handler
+	}
+
+	OutlierNamespace struct {
+		Message string
 	}
 )
 
 func NewEagerNamespaceRefresher(
 	metadataManager persistence.MetadataManager,
 	namespaceRegistry namespace.Registry,
-	logger log.Logger) EagerNamespaceRefresher {
+	logger log.Logger,
+	clientBean client.Bean,
+	replicationTaskExecutor namespace.ReplicationTaskExecutor,
+	currentCluster string,
+	metricsHandler metrics.Handler) EagerNamespaceRefresher {
 	return &eagerNamespaceRefresherImpl{
-		metadataManager:   metadataManager,
-		namespaceRegistry: namespaceRegistry,
-		logger:            logger,
+		metadataManager:         metadataManager,
+		namespaceRegistry:       namespaceRegistry,
+		logger:                  logger,
+		clientBean:              clientBean,
+		replicationTaskExecutor: replicationTaskExecutor,
+		currentCluster:          currentCluster,
+		metricsHandler:          metricsHandler,
 	}
 }
 
+func (e *OutlierNamespace) Error() string {
```

Comment:
what about using existing FailedPrecondition error?

Reply:
Sure, did not realize we have this error type already.